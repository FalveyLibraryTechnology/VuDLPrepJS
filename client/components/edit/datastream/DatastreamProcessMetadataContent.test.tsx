import React from "react";
import { describe, afterEach, expect, it, jest } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import DatastreamProcessMetadataContent from "./DatastreamProcessMetadataContent";
import { waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";

const mockUseGlobalContext = jest.fn();
jest.mock("../../../context/GlobalContext", () => ({
    useGlobalContext: () => {
        return mockUseGlobalContext();
    },
}));

const mockUseEditorContext = jest.fn();
jest.mock("../../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

const mockUseProcessMetadataContext = jest.fn();
jest.mock("../../../context/ProcessMetadataContext", () => ({
    useProcessMetadataContext: () => {
        return mockUseProcessMetadataContext();
    },
}));

const mockUseDatastreamOperation = jest.fn();
jest.mock("../../../hooks/useDatastreamOperation", () => () => {
    return mockUseDatastreamOperation();
});

let mockDatastreamProcessMetadataTaskProps = [];
const DatastreamProcessMetadataTask = function DatastreamProcessMetadataTask(props) {
    mockDatastreamProcessMetadataTaskProps.push(props);
    return "";
};
jest.mock("./DatastreamProcessMetadataTask", () => DatastreamProcessMetadataTask);

jest.mock("@mui/x-date-pickers", () => ({
    DateTimePicker: () => "DateTimePicker",
    LocalizationProvider: () => "LocalizationProvider",
}));

jest.mock("@mui/material/Box", () => (props) => props.children);
jest.mock("@mui/material/FormControl", () => (props) => props.children);
jest.mock("@mui/material/FormLabel", () => (props) => props.children);
let tabChangeFunction: ((tab: number) => void) | null = null;
jest.mock("@mui/material/Tabs", () => (props) => {
    tabChangeFunction = props.onChange;
    return props.children;
});
jest.mock("@mui/material/Tab", () => (props) => `Tab: ${props.label}`);
jest.mock("@mui/material/Grid", () => (props) => props.children);
let pidPickerFunction: ((pid: string) => void) | null = null;
jest.mock("../PidPicker", () => (props) => {
    pidPickerFunction = props.setSelected;
    return "PidPicker: " + JSON.stringify(props);
});

describe("DatastreamProcessMetadataContent", () => {
    let datastreamOperationValues;
    let editorContext;
    let globalValues;
    let processMetadataValues;

    const renderComponent = async (fakeData = {}) => {
        datastreamOperationValues.getProcessMetadata.mockResolvedValue(fakeData);
        processMetadataValues.state = fakeData;

        render(<DatastreamProcessMetadataContent />);

        await waitFor(() => expect(processMetadataValues.action.setMetadata).toHaveBeenCalledWith(fakeData));
    };

    const getRenderedTree = async (fakeData = {}, extraStep: (() => void) | null = null) => {
        datastreamOperationValues.getProcessMetadata.mockResolvedValue(fakeData);
        processMetadataValues.state = fakeData;

        const tree = renderer.create(<DatastreamProcessMetadataContent />);

        await renderer.act(async () => {
            await waitFor(() => expect(processMetadataValues.action.setMetadata).toHaveBeenCalledWith(fakeData));
        });

        if (extraStep) {
            await renderer.act(async () => {
                extraStep();
            });
        }
        return tree.toJSON();
    };

    beforeEach(() => {
        editorContext = {
            state: {
                childListStorage: {},
                objectDetailsStorage: {},
            },
            action: {
                getChildListStorageKey: jest.fn(),
                loadChildrenIntoStorage: jest.fn(),
                loadObjectDetailsIntoStorage: jest.fn(),
            },
        };
        mockUseEditorContext.mockReturnValue(editorContext);
        datastreamOperationValues = {
            uploadProcessMetadata: jest.fn(),
            getProcessMetadata: jest.fn(),
        };
        mockUseDatastreamOperation.mockReturnValue(datastreamOperationValues);
        globalValues = { action: { closeModal: jest.fn() } };
        mockUseGlobalContext.mockReturnValue(globalValues);
        mockDatastreamProcessMetadataTaskProps = [];
        processMetadataValues = {
            state: {},
            action: {
                addTask: jest.fn(),
                deleteTask: jest.fn(),
                setMetadata: jest.fn(),
                setProcessCreator: jest.fn(),
                setProcessDateTime: jest.fn(),
                setProcessLabel: jest.fn(),
                setProcessOrganization: jest.fn(),
                updateTaskAttributes: jest.fn(),
            },
        };
        mockUseProcessMetadataContext.mockReturnValue(processMetadataValues);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("renders a loading message if content is unavailable", () => {
        datastreamOperationValues.getProcessMetadata.mockResolvedValue({});
        const tree = renderer.create(<DatastreamProcessMetadataContent />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders a form when empty data is loaded", async () => {
        const tree = await getRenderedTree();
        expect(processMetadataValues.action.addTask).toHaveBeenCalledWith(0);
        expect(tree).toMatchSnapshot();
    });

    it("supports tab switching", async () => {
        const tree = await getRenderedTree({}, () => {
            if (tabChangeFunction) {
                tabChangeFunction(null, 1);
            }
        });
        expect(processMetadataValues.action.addTask).toHaveBeenCalledWith(0);
        expect(tree).toMatchSnapshot();
    });

    it("renders a form when non-empty data is loaded", async () => {
        const tree = await getRenderedTree({
            processLabel: "label",
            processCreator: "creator",
            processDateTime: "datetime",
            processOrganization: "organization",
            tasks: [{ id: 1 }, { id: 2 }],
        });
        expect(processMetadataValues.action.addTask).not.toHaveBeenCalled();
        expect(tree).toMatchSnapshot();
    });

    it("has a working save button", async () => {
        await renderComponent();
        await userEvent.setup().click(screen.getByText("Save"));
        expect(datastreamOperationValues.uploadProcessMetadata).toHaveBeenCalledWith(processMetadataValues.state);
    });

    it("ignores empty PIDs in clone input", async () => {
        await renderComponent();
        expect(pidPickerFunction).not.toBeNull();
        await act(async () => {
            pidPickerFunction && pidPickerFunction("");
        });
        expect(editorContext.action.loadObjectDetailsIntoStorage).not.toHaveBeenCalled();
    });

    it("handles errors that occur during PID cloning", async () => {
        await renderComponent();
        const alertSpy = jest.spyOn(window, "alert").mockImplementation(jest.fn());
        editorContext.action.loadObjectDetailsIntoStorage.mockImplementation(
            (pid: string, errorCallback: () => void) => {
                expect(pid).toEqual("foo:123");
                errorCallback();
            },
        );
        expect(pidPickerFunction).not.toBeNull();
        await act(async () => {
            pidPickerFunction && pidPickerFunction("foo:123");
        });
        expect(alertSpy).toHaveBeenCalledWith("Cannot load PID: foo:123");
    });

    it("supports loading PID data into storage after selection for cloning", async () => {
        await renderComponent();
        expect(pidPickerFunction).not.toBeNull();
        await act(async () => {
            pidPickerFunction && pidPickerFunction("foo:123");
        });
        expect(editorContext.action.loadObjectDetailsIntoStorage).toHaveBeenCalledWith("foo:123", expect.anything());
    });

    it("validates datastreams before cloning metadata from another PID", async () => {
        editorContext.state.objectDetailsStorage = {
            "foo:123": {
                datastreams: [],
            },
        };
        const alertSpy = jest.spyOn(window, "alert").mockImplementation(jest.fn());
        await renderComponent();
        expect(pidPickerFunction).not.toBeNull();
        await act(async () => {
            pidPickerFunction && pidPickerFunction("foo:123");
        });
        await userEvent.setup().click(screen.getByText("Clone"));
        expect(alertSpy).toHaveBeenCalledWith("foo:123 does not contain a PROCESS-MD datastream.");
    });

    it("can clone metadata from another PID", async () => {
        editorContext.state.objectDetailsStorage = {
            "foo:123": {
                datastreams: ["PROCESS-MD"],
            },
        };
        await renderComponent();
        expect(pidPickerFunction).not.toBeNull();
        await act(async () => {
            pidPickerFunction && pidPickerFunction("foo:123");
        });
        await userEvent.setup().click(screen.getByText("Clone"));
        expect(datastreamOperationValues.getProcessMetadata).toHaveBeenCalledWith("foo:123", true);
    });

    it("supports task updates", async () => {
        await renderComponent({ tasks: [{ id: 1 }] });
        const attr = { foo: "bar" };
        mockDatastreamProcessMetadataTaskProps[0].setAttributes(attr, true);
        expect(processMetadataValues.action.updateTaskAttributes).toHaveBeenCalledWith(0, attr);
    });

    it("supports adding tasks", async () => {
        await renderComponent({ tasks: [{ id: 1 }] });
        mockDatastreamProcessMetadataTaskProps[0].addBelow();
        expect(processMetadataValues.action.addTask).toHaveBeenCalledWith(1);
    });

    it("supports deleting tasks", async () => {
        await renderComponent({ tasks: [{ id: 1 }] });
        mockDatastreamProcessMetadataTaskProps[0].deleteTask();
        expect(processMetadataValues.action.deleteTask).toHaveBeenCalledWith(0);
    });

    it("can be canceled", async () => {
        await renderComponent();
        await userEvent.setup().click(screen.getByText("Cancel"));
        expect(datastreamOperationValues.uploadProcessMetadata).not.toHaveBeenCalled();
        expect(globalValues.action.closeModal).toHaveBeenCalledWith("datastream");
    });
});
