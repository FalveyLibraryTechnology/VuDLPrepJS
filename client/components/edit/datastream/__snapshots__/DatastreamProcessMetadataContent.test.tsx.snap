// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamProcessMetadataContent renders a form when empty data is loaded 1`] = `
[
  <div
    className="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    Digital Provenance
    Tab: Editor
    Tab: Clone
    <div
      style={
        {
          "display": "block",
        }
      }
    >
      Process Label
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r3:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Process Label
            </span>
          </legend>
        </fieldset>
      </div>
      Process Creator
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r4:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Process Creator
            </span>
          </legend>
        </fieldset>
      </div>
      LocalizationProvider
      Process Organization
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r5:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Process Organization
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        onClick={[Function]}
      >
        Add Task
      </button>
    </div>
    <div
      style={
        {
          "display": "none",
        }
      }
    >
      <div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
            >
              Enter PID manually
            </div>
            <div
              className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style={
              {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  className="MuiAccordion-region"
                  role="region"
                >
                  <div
                    className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <label>
                      Enter PID:
                       
                      <input
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                    </label>
                    <button
                      onClick={[Function]}
                    >
                      Set
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
            >
              Choose PID from Tree
            </div>
            <div
              className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style={
              {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  className="MuiAccordion-region"
                  role="region"
                >
                  <div
                    className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <p>
                      <span
                        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
                        role="progressbar"
                        style={
                          {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                      >
                        <svg
                          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                          viewBox="22 22 44 44"
                        >
                          <circle
                            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                            cx={44}
                            cy={44}
                            fill="none"
                            r={20.2}
                            strokeWidth={3.6}
                            style={{}}
                          />
                        </svg>
                      </span>
                       Loading...
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary uploadProcessMetadataButton css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Save
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Cancel
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;

exports[`DatastreamProcessMetadataContent renders a form when non-empty data is loaded 1`] = `
[
  <div
    className="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    Digital Provenance
    Tab: Editor
    Tab: Clone
    <div
      style={
        {
          "display": "block",
        }
      }
    >
      Process Label
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r6:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value="label"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Process Label
            </span>
          </legend>
        </fieldset>
      </div>
      Process Creator
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r7:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value="creator"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Process Creator
            </span>
          </legend>
        </fieldset>
      </div>
      LocalizationProvider
      Process Organization
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r8:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value="organization"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Process Organization
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      style={
        {
          "display": "none",
        }
      }
    >
      <div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
            >
              Enter PID manually
            </div>
            <div
              className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style={
              {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  className="MuiAccordion-region"
                  role="region"
                >
                  <div
                    className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <label>
                      Enter PID:
                       
                      <input
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                    </label>
                    <button
                      onClick={[Function]}
                    >
                      Set
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
            >
              Choose PID from Tree
            </div>
            <div
              className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style={
              {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  className="MuiAccordion-region"
                  role="region"
                >
                  <div
                    className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <p>
                      <span
                        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
                        role="progressbar"
                        style={
                          {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                      >
                        <svg
                          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                          viewBox="22 22 44 44"
                        >
                          <circle
                            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                            cx={44}
                            cy={44}
                            fill="none"
                            r={20.2}
                            strokeWidth={3.6}
                            style={{}}
                          />
                        </svg>
                      </span>
                       Loading...
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary uploadProcessMetadataButton css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Save
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Cancel
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;

exports[`DatastreamProcessMetadataContent renders a loading message if content is unavailable 1`] = `
<div
  className="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
>
  Loading...
</div>
`;
